docs/content/integrations/snowflake/reference.mdx

```diff
@@ -74,7 +114,62 @@ defs = Definitions(
 
 ---
 
-## Selecting specific columns in a downstream asset
+## Using the Snowflake resource
+
+### Executing custom SQL commands
+
+Using a [Snowflake resource](/concepts/resources), you can execute custom SQL queries on a Snowflake database:
+
+```python file=/integrations/snowflake/resource.py startafter=start endbefore=end
+import pandas as pd
+from dagster_snowflake import SnowflakeResource
+
+from dagster import Definitions, EnvVar, asset
+
+# this example executes a query against the IRIS_DATASET table created in Step 2 of the
+# Using Dagster with Snowflake tutorial
+
+
+@asset
+def small_petals(snowflake: SnowflakeResource) -> pd.DataFrame:
+    with snowflake.get_connection() as conn:
+        return (
```

Comment:
this example still uses an IO Manager (it returns a pd.DataFrame). I wonder if maybe we should alter the example so that it returns a MaterializeResult with the number of rows retrieved or some other summary?

Reply:
+1 to this

Reply:
the sql should probably be updated to create the SMALL_PETALS table too